web/admin/src/App.js

```diff
@@ -448,12 +448,16 @@ class App extends PureComponent {
   }
 
   findCompletedCategoryTime = (completedScans, cat, allScans) => {
+    const filteredScanKeys = Object.keys(allScans).filter(scanId => {
```

Comment:
Instead of keeping a list of keys (which you don't directly need) and constantly translating them into the actual scans, you can keep the scans themselves:

```js
const filteredScans = Object.entries(allScans)
  .filter(([scanId, scan] => ...)
  .map(([_, scan]) => scan)
```